src/coins.cpp

```diff
@@ -325,14 +322,14 @@ void CCoinsViewCache::SanityCheck() const
         if (entry.IsDirty()) attr |= 1;
         if (entry.IsFresh()) attr |= 2;
         if (entry.coin.IsSpent()) attr |= 4;
-        // Only 5 combinations are possible.
-        assert(attr != 2 && attr != 4 && attr != 7);
+        // Only 4 combinations are possible.
```

Comment:
A coin cannot be spent and not DIRTY. I made a new commit to address this in AddCoin. The only time it will be in this state is when a `CCoinsViewCacheEntry` is constructed with an empty coin, which then has the coin moved into it a few lines later.

I think doing a follow up where we remove direct access to flags can remove a lot of code paths that we test for.

Reply:
That makes sense, as stated in other comments, I would be more comfortable doing that before this refactoring.